packages/shortcode/src/index.js

```diff
@@ -102,7 +102,7 @@ export function replace( tag, text, callback ) {
 
 		// Make sure to return any of the extra brackets if they weren't used to
 		// escape the shortcode.
-		return result ? left + result + right : match;
+		return typeof result === 'string' ? left + result + right : match;
```

Comment:
Hi @dsifford thank you for your contribution. This change fixes the problem, but I noticed that if the callback returned a number previously the replace happened and with this change the replace is not happening. I added a test case for the scenario.
What if we use the condition `result || result === ''`?

Reply:
Up to you guys. I'd be more of a fan of making the return type more strict, but I'll concede to the majority. Non-strict return types can lead to some tricky bugs.

Reply:
I'd be curious what the usage looks like for someone who wants to opt out of the concatenation. In other words, if that usage should expect to provide a return value of `undefined` or `null` from the callback, couldn't we form the condition to target those values, which has the benefit of not breaking existing usage with values like numbers, while also type-specific to "empty" (skip) case.

Since I'd not really trust the result of `+` and potentially-numeric values (could add rather than concatenate?), I'd also not have too much of a problem with saying that the previous behavior was undocumented / unintended / buggy, calling this a "fix".